src/services/pipewire/node.hpp

```diff
@@ -86,10 +86,46 @@ class PwAudioChannel: public QObject {
 	/// including aux and custom channel ranges.
 	Q_INVOKABLE static QString toString(qs::service::pipewire::PwAudioChannel::Enum value);
 };
+///! The type of a pipewire node. Use bitwise comparisons to
+/// filter for audio, video, sink, source or stream nodes
+class PwNodeType: public QObject {
+	Q_OBJECT;
+	QML_ELEMENT;
+	QML_SINGLETON;
 
-enum class PwNodeType : quint8 {
-	Untracked,
-	Audio,
+public:
+	enum Enum : quint8 {
+		// A Pipewire node which is not being managed.
+		Untracked = 0x1,
+		// This flag is set when this node is an Audio node.
+		Audio = 0x2,
+		// This flag is set when this node is an Video node.
+		Video = 0x4,
+		// This flag is set when this node is a stream node.
+		Stream = 0x8,
+		// This flag is set when this node is producing some form of data,
+		// such as a microphone, screenshare or webcam.
+		Source = 0x10,
+		// This flag is set when this node is receiving data.
+		Sink = 0x20,
+		// A sink for audio samples, like an audio card.
+		AudioSink = Audio | Sink,
+		// A source of audio samples like a microphone.
+		AudioSource = Audio | Source,
+		// A node that is both a sink and a source.
+		AudioDuplex = Audio | Sink | Source,
+		// A playback stream.
+		AudioOutStream = Audio | Sink | Stream,
+		// A capture stream.
+		AudioInStream = Audio | Source | Stream,
+		// A producer of video, like a webcam or a screenshare.
+		VideoSource = Video | Source,
+		// A consumer of video, such as a program that is recieving a video stream.
+		VideoSink = Video | Sink,
```

Comment:
Note which pipewire media class each of these is, and what they're equal to, as the definitions don't currently show up in docs.